<h2>Description</h2>
<p>
Manually curated MHC Class I and Class II genes.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
Each gene is colored accordingly as a block with directions. Color code follows that of Extended Data Fig. 3 from Yoo et al. Species name is added to each gene family in the annotation (e.g. Gogo for <i>Go</i>rilla <i>go</i>rilla). <br><br>
</p>

<h3>MHC Class I</h3>
<ul>
  <li><span style=background-color:#7F7DBB;>&nbsp;&nbsp;&nbsp;</span> - F
  <li><span style=background-color:#89CCED;>&nbsp;&nbsp;&nbsp;</span> - G
  <li><span style=background-color:#44AA99;>&nbsp;&nbsp;&nbsp;</span> - H
  <li><span style=background-color:#85C887;>&nbsp;&nbsp;&nbsp;</span> - A
  <li><span style=background-color:#5590CC;>&nbsp;&nbsp;&nbsp;</span> - J
  <li><span style=background-color:#DDCB77;>&nbsp;&nbsp;&nbsp;</span> - E
  <li><span style=background-color:#CC6677;>&nbsp;&nbsp;&nbsp;</span> - C
  <li><span style=background-color:#D375A9;>&nbsp;&nbsp;&nbsp;</span> - B
  <li><span style=background-color:#4FC1B4;>&nbsp;&nbsp;&nbsp;</span> - OKO</li>
</ul>
<h3>MHC Class II</h3>
<ul>
  <li><span style=background-color:#8CD4E2;>&nbsp;&nbsp;&nbsp;</span> - DRA</li>
  <li><span style=background-color:#B0B06A;>&nbsp;&nbsp;&nbsp;</span> - DRB3</li>
  <li><span style=background-color:#B49DCB;>&nbsp;&nbsp;&nbsp;</span> - DRB1</li>
  <li><span style=background-color:#CC7A8D;>&nbsp;&nbsp;&nbsp;</span> - DRB5</li>
  <li><span style=background-color:#77ADDE;>&nbsp;&nbsp;&nbsp;</span> - DQA1</li>
  <li><span style=background-color:#3F81B8;>&nbsp;&nbsp;&nbsp;</span> - DQB1</li>
  <li><span style=background-color:#279E48;>&nbsp;&nbsp;&nbsp;</span> - DQA2</li>
  <li><span style=background-color:#E8FF52;>&nbsp;&nbsp;&nbsp;</span> - DQB2</li>
  <li><span style=background-color:#F6EB14;>&nbsp;&nbsp;&nbsp;</span> - DOB</li>
  <li><span style=background-color:#FBB471;>&nbsp;&nbsp;&nbsp;</span> - DMB</li>
  <li><span style=background-color:#B89BC9;>&nbsp;&nbsp;&nbsp;</span> - DMA</li>
  <li><span style=background-color:#DC9FC7;>&nbsp;&nbsp;&nbsp;</span> - DOA</li>
  <li><span style=background-color:#DECBE4;>&nbsp;&nbsp;&nbsp;</span> - DPA1</li>
  <li><span style=background-color:#665FAA;>&nbsp;&nbsp;&nbsp;</span> - DPB1</li>
</ul>
<h3>Pseudogenes</h3>
All pseudogenes are colored in grey despite its classified gene name.
<ul>
  <li><span style=background-color:#B4B4B4;>&nbsp;&nbsp;&nbsp;</span> - Pseudogene</li>
</ul>

<h2>Methods</h2>
<p>
  We considered the “MHC genomic region” of each ape genomic assembly to lie between the genes <i>GABBR1</i> and <i>KIFC1</i>, as previously determined by Shiina et al. (2017). The chromosomal locations of each ape MHC genomic region were located by first aligning each respective haplotype of each T2T ape assembly to the human T2T assembly (NCBI assembly T2T-CHM13v2.0) as a reference with minimap2 v2.28 (Li, 2018). SAMtools (Danecek et al., 2021) was then used to filter (with <code>-F4</code> flag) and sort SAM files resulting from running minimap2. Sorted SAM files were further converted into BAM format and indexed with SAMtools. Then we subset regions of each ape assembly haplotype that mapped human chromosome 6 between coordinates 2937649-3331258, which flank the MHC region of human T2T assembly T2T-CHM13v2.0, with the SAMtools view subcommand.
<ul>
<p>
  To annotate putative classical and nonclassical MHC class I and class II genes for both haplotypes of each ape T2T genomic assembly, two complementary
approaches were used. First, we ran EXONERATE 2.4 113 with the <code>est2genome</code> mapping model
(https://www.ebi.ac.uk/about/vertebrate-genomics/software/exonerate) recursively with functional human HLA gene and CDS annotations from T2T-CHM13v2.0 and
the IPD-MHC database (Release 3.12.0.0 build 211; https://www.ebi.ac.uk/ipd/mhc/) gene and
CDS annotations for chimpanzee, bonobo, western lowland gorilla, Sumatran orangutang, and Bornean orangutang. EXONERATE results were filtered to matches of greater than 95%.
</p>
<p>
Second, we mapped human HLA gene annotations to chromosome 23 of siamang and
chromosome 5 for all remaining ape species with minimap2. Results between EXONERATE
and minimap2 were compared manually for concordance. Then annotations were further manually verified and curated with ALIVIEW 2.8 (Larson, 2014), and all gene annotations were individually compared against all available human and ape species orthologs to confirm proper annotation. Due to the absence of previous gene and CDS annotations for the siamang, we assigned MHC gene annotations for presumed start codon to stop codon based on corresponding human HLA gene and CDS annotations. MHC class I and class II gene names were assigned according to human and ape species orthologs.
</p>
<p>
Phylogenetic trees were produced for MHC class I loci and MHC class II DRB loci to confirm the identity of MHC genes exhibiting copy number variation. First, separate MHC class I and MHC-DRB multiple sequence alignments were produced online with MAFFT (https://mafft.cbrc.jp/alignment/server/index.html). Both alignments consisted of genomic sequence of annotated MHC genes from ape T2T assemblies, human coding sequence (CDS) and genomic sequence of genes from T2T-CHM13v2.0. Genomic and coding sequences for MHC class II DRB genes and MHC class I loci were obtained from the IPD-MHC database.
</p>
<p>
To generate phylogenetic trees from sequences from all MHC class I and the MHC class II DRB genes, respectively, the IQ-TREE online (http://iqtree.cibiv.univie.ac.at) was used with default settings (Trifinopoulos et al., 2016). IQ-TREE automatically selected the best-fitting substitution models and calculated node support with Ultrafast Bootstrap (Hoang et al., 2018). Resulting phylogenetic trees from IQ-TREE were plotted using RStudio (version 2024.04.2+764) with the ggtree (Yu et al., 2017) and ape (Paradis and Schilep, 2019) packages, retaining bootstrap values above 70.
</p>
<p>
The results from this manually curated annotation approach were compared to MHC annotations of the automated genome-wide CAT annotation pipeline. Since these two approaches use different criteria and evidence for annotation, minor discrepancies are expected. We reached a consensus between the two approaches based on phylogenetic results, genomic coordinate concordance in terms of genomic location and length obtained by both approaches for each locus, as well as presence/absence of each annotated locus according to each annotation approach.
</p>
</ul>
</p>
<h2>Credits</h2>
<p>
Original annotation was prepared by Joanna Malukiewicz and Tobias Lenz at the University of Hamburg.<br>
The tracks were adjusted to be displayed by Arang Rhie at NHGRI, NIH.
</p>
<h2>References</h2>
<p>
<li>Yoo, D., Rhie, A., Hebbar, P. <i>et al</i>. Complete sequencing of ape genomes. <i>bioRxiv</i> (2024) <a href="https://doi.org/10.1101/2024.07.31.605654" target="_blank">doi:10.1101/2024.07.31.605654</a></li>
<li>Danecek, P. et al. Twelve years of SAMtools and BCFtools. Gigascience 10, giab008 (2021).</li>
<li>Hoang, D. et al. UFBoot2: improving the ultrafast bootstrap approximation. Molecular biology and evolution 35, 518-522 (2018).</li>
<li>Larsson, A. AliView: a fast and lightweight alignment viewer and editor for large data sets. Bioinformatics30(22): 3276-3278. (2014).</li>
<li>Li, H. Minimap2: pairwise alignment for nucleotide sequences. Bioinformatics 34, 3094-3100 (2018).</li>
<li>Paradis, E. & Schliep, K. ape 5.0: an environment for modern phylogenetics and evolutionary analyses in R. Bioinformatics 35, 526-528 (2019).</li>
<li>Shiina, T. et al. Comparative genomics of the human, macaque and mouse major histocompatibility complex. Immunology 150, 127-138 (2017).</li>
<li>Trifinopoulos, J. et al. IQ-TREE: a fast online phylogenetic tool for maximum likelihood analysis. Nucleic acids research 44, W232-W235 (2016).</li>
<li>Yu, G. et al. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution 8, 28-36 (2017).</li>
</p>
