<h2>Description</h2>

<p>This annotation track for the HG002 Q100 assembly was generated by the Salzberg Lab at Johns Hopkins University using <a href="https://github.com/agshumate/Liftoff">Liftoff</a>. The track includes both coding and non-coding genomic features (e.g., genes, transcripts, exons, and CDS) lifted over from the T2T CHM13 assembly onto the HG002 Q100 v1.1 assembly, hereinafter referred to as the Q100 assembly.</p>

<h2>Methods</h2>

<p>Annotating a haplotype-resolved diploid assembly differs from annotating a pseudo-haploid assembly, as it involves treating each haplotype as a separate genome. Specifically, LiftOff was run separately on the complete set of maternal haplotypes, followed by a second run on the complete set of paternal haplotypes of the Q100 assembly. All steps described below were repeated for each haplotype. All scripts used to generate this annotation are available <a href="https://github.com/haydenji0731/hg002-q100-annotation">here</a>.</p>

<h3>Two-pass LiftOff</h3>

<p>LiftOff was run in two passes. In the first pass, features were lifted over from the <a href="https://github.com/marbl/CHM13?tab=readme-ov-file">JHU RefSeqv110 + LiftOff v5.2 annotation</a> of the T2T CHM13 assembly. The following second pass lifts over the rDNA arrays, each consisting of three consecutive rDNA genes (18S, 5.8S, and 28S), with a precursor 45S rDNA gene spanning the entire array.</p>

<p>Priot to the first pass, a 44,838 bp reference 45S rDNA sequence (Kim et al., 2018) was mapped to the Q100 assembly using <a href="https://github.com/mummer4/mummer">nucmer</a>. Regions that aligned with >1,000 bp and >98.5% sequence identity were masked to prevent LiftOff from annotating features in these putative rDNA array regions. Additionally, putative V(D)J regions, whose coordinates were provided by Nathan Dwarshuis at NIST, were also masked. LiftOff was then used to lift features from the CHM13 reference annotation (excluding V(D)J and rDNA array features) onto the masked Q100 assembly, constituting the first pass.</p>

<p>In the second pass, rDNA annotations from CHM13 were lifted over to the unmasked, full Q100 assembly.</p>

<p>After completing both passes, the annotations resulting from each pass were first intersected to ensure that no gene loci overlapped. Then, the two annotations were merged and sorted by coordinates using <a href="https://ccb.jhu.edu/software/stringtie/gff.shtml">gffread</a>.</p>

<h3>Post-hoc CDS Polishing</h3>

<p>While LiftOff maximizes sequence-level identities between the reference and newly annotated genes, it sometimes makes errors when annotating CDS features. These errors often arise when the gene or transcript sequences diverge from the reference. To improve the quality of CDS annotations, the following was enforced:</p>

<ul>
  <li><em>CDS lengths should be divisible by three.</em> When a CDS is broken (e.g., due to an invalid start or stop codon, or a premature stop codon), LiftOff often annotates the CDS with 1–2 extra base pairs at the start or end. To resolve this, bases were removed from the end if the CDS has a valid start codon, or from either end, whichever results in higher sequence similarity with the corresponding reference gene after trimming.</li>
  <li><em>Alternative upstream start and/or downstream stop codons are considered valid.</em> CDSs with an invalid start codon are extended upstream to look for an alternative valid start. Similarly, for CDSs with an invalid stop codon, extensions are made downstream. As a result, some CDSs will encode a longer protein than the original LiftOff annotation, which is likely to be valid but not accounted for by the reference. CDSs with an in-frame stop codon were excluded from this search.</li>
</ul>

<h3>Tags and IDs</h3>

<p>The primary goal of this annotation is to assign unique IDs to genes and transcripts in the Q100 assembly, with IDs that refer specifically to entities within this assembly. Each gene and transcript is assigned a unique ID formatted as follows:</p>

<ul>
  <li>gene_id format: hg002_[chr]_[maternal/paternal]_[gene_ordinal]</li>
  <li>transcript_id format: hg002_[chr]_[maternal/paternal]_[gene_ordinal].[transcript_ordinal]</li>
</ul>

<p>Here, <em>gene_ordinal</em> refers to the ordinal number of a gene within the same chromosome, where genes are enumerated from the 5' to 3' direction. <em>transcript_ordinal</em> refers to the ordinal number of a transcript within its parent gene locus, where transcripts are enumerated randomly.</p>

<p>In addition to standard tags (e.g., Parent, ID, biotype), this annotation also includes tags specific to protein-coding genes and transcripts, such as: "valid_ORFs", "matches_ref_protein", "valid_ORF", "missing_start_codon", "missing_stop_codon", and "inframe_stop_codon".</p>

<h2>Credits</h2>

<p>Hyun Joo Ji (Johns Hopkins University)</p>
<p>Kuan-hao Chao (Johns Hopkins University)</p>
<p>Steven L. Salzberg (Johns Hopkins University)</p>

<p>For questions, please contact <a href="mailto:hji20@jh.edu">hji20@jh.edu</a>.</p>

<h2>References</h2>

<p>Nurk, S., Koren, S., Rhie, A., Rautiainen, M., Bzikadze, A. V., Mikheenko, A., ... & Phillippy, A. M. (2022). The complete sequence of a human genome. <em>Science</em>, <strong>376</strong>(6588), 44–53. </p>
<p>Shumate, A., & Salzberg, S. L. (2021). Liftoff: Accurate mapping of gene annotations. <em>Bioinformatics</em>, <strong>37</strong>(12), 1639–1643.</p>
<p>Kim, J. H., Dilthey, A. T., Nagaraja, R., Lee, H. S., Koren, S., Dudekula, D., ... & Larionov, V. (2018). Variation in human chromosome 21 ribosomal RNA genes characterized by TAR cloning and long-read sequencing. <em>Nucleic Acids Research</em>, <strong>46</strong>(13), 6712–6725.</p>
<p>Pertea, G., & Pertea, M. (2020). GFF utilities: GffRead and GffCompare. <em>F1000Research</em>, <strong>9</strong>, ISCB-Comm.</p>
<p>Marçais, G., Delcher, A. L., Phillippy, A. M., Coston, R., Salzberg, S. L., & Zimin, A. (2018). MUMmer4: A fast and versatile genome alignment system. PLoS computational biology, 14(1), e1005944.</p>
