<h2>Description</h2>
These tracks display nucleotide frequencies generated by Mitchell Vollger's
<a href="https://github.com/mrvollger/NucFreq">NucFreq</a> program run on 
Winnowmap2 primary alignments of HiFi Revio reads called with DeepConsensus v1.2
or the ONT Epi2Me release of "Q28" reads v1.1 HG002 assembly. The
input bam files are available <a href="https://s3-us-west-2.amazonaws.com/human-pangenomics/index.html?prefix=T2T/HG002/assemblies/polishing/HG002/v1.1/mapping/hifi_revio_pbmay24/">here</a> and <a href="https://s3-us-west-2.amazonaws.com/human-pangenomics/index.html?prefix=T2T/HG002/assemblies/polishing/HG002/v1.1/mapping/ont_epi2me_q28/">here</a>.

<h2>Display Conventions and Configuration</h2>

<p>
The counts displayed in black are counts of the most common base call in reads
aligned at a given position. The counts in red are for the second most common
base call in reads aligned at the same position.
</p>

<h2>Credits</h2>
<p>
Tracks were prepared by Nancy F Hansen. The NucFreq program was written by Mitchell R Vollger.
</p>

<h2>References</h2>
<p>
Vollger MR, Dishuck PC, Sorensen M, Welch AE, Dang V, Dougherty ML, <em>et al</em>. Long-read sequence and assembly of segmental duplications. Nat Methods. 2019;16: 88â€“94. doi:10.1038/s41592-018-0236-3
</p>


