<h2>Description</h2>
<b>F</b>iber-seq <b>I</b>nferred <b>R</b>egulatory <b>E</b>lements

<p>
These tracks represent the <a href="https://github.com/fiberseq/FIRE" target="_blank">FIRE</a> peak calls, the inferred regulatory elements from Fiber-seq.
</p>

<p>
	FIREs are MTase sensitive patches (MSPs) that are inferred to be regulatory elements on single chromatin fibers. To do this we used semi-supervised machine learning to identify MSPs that are likely to be regulatory elements using the <code>Mokapot</code> framework and <code>XGBoost</code>. Every individual FIRE element is associated with a precision value, which indicates the probability that the FIRE element is a true regulatory element. The precision of FIREs elements are estimated using <code>Mokapot</code> and validation data not used in training. We train our model targeting FIRE elements with at least 90% precision, MSPs with less than 90% precision are considered to have average level of accessibility expected between two nucleosomes, and are referred to as linker regions.
	
	Semi-superivized machine learning with <code>Mokapot</code> requires a mixed-positive training set and a clean negative training set. To create mixed positive training data we selected MSPs that overlapped DNase hypersensitive sites (DHSs) and CTCF ChIP-seq peaks. And to create a clean negative training set we selected MSPs that did not overlap DHSs or CTCF ChIP-seq peaks.
</p>
<p>
	<li><u>CHM13 FIRE peaks</u>: Peaks are called by identifying FIRE score local-maxima that have FDR values below a threshold. By default the pipeline reports results for both 1% and 5% FDR thresholds. Once a local-maxima is identified, the start and end positions of the peak are determined by the median start and end positions of the underlying FIRE elements. We also calculate and report wide peaks by taking the union of the FIRE peaks and all regions below the FDR threshold and then merging resulting regions that are within one nucleosome (147 bp) of one another.

	<li><u>CHM13 FIRE peak FDR</u>: FDR calculation begins by shuffling the locations of all the fibers across the genome and recalculating the FIRE score for each position in the genome. The FDR is then defined as the number of bases that have shuffled FIRE scores above a threshold divided by the number of bases in the un-shuffled data.
</p>

<h2>Methods</h2>
<p>Manuscript in preparation. Please refer to <a href="https://github.com/fiberseq/FIRE/tree/main/docs" target="_blank">https://github.com/fiberseq/FIRE/tree/main/docs</a> for more details.

<h2>Credits</h2>
Tracks were generated by Mitchell Vollger (mvollger_at_uw.edu) and Andrew Stergachis (absterga_at_uw.edu).

<h2>References</h2>
<p>
Stergachis <em>et al</em>, Single-molecule regulatory architectures captured by chromatin fiber sequencing. Science 2022 <a href="DOI: 10.1126/science.aaz1646" target="_blank">DOI: 10.1126/science.aaz1646</a>
</p>


