<h2>Description</h2>
<p>
Gene models for the assembly generated using the CAT (Compartive Annotation Toolkit) software using transMap, Liftoff, miniprot, and AUGUSTUS modules.<br>
The reference annotations are from the T2T-CHM13v2 Curated RefSeq v5.2 annotations.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
This track follows the display conventions for gene prediction tracks.<br>
The exons for putative non-coding genes and untranslated regions are represented by relatively thin blocks, while those for coding open reading frames are thicker.<br>
Gene names are displayed in 'pack' or 'full' mode. More information about each gene can be found by clicking on the specific gene/transcript model.<br><br>

The following color key is used:
</p>
<ul>
<li><span style='background-color:#0ABEFE;'>&nbsp;&nbsp;&nbsp;</span> - orthologs mapped using transMap
<li><span style='background-color:#008000;'>&nbsp;&nbsp;&nbsp;</span> - orthologs and paralogs mapped using miniprot
<li><span style='background-color:#FFC0CB;'>&nbsp;&nbsp;&nbsp;</span> - orthologs and paralogs mapped using Liftoff
</ul>
<h2>Methods</h2>
<p>
This tracks displays gene annotations generated by multiple modules in the Comparative Annotation Toolkit (CAT).
<ul>
<h3>Comparative Annotation Toolkit 1</h3>
Genome annotation for the assembly was performed using Comparative Annotation Toolkit (CAT).<br>
CAT leverages whole-genome alignments generated by Cactus to transfer annotations from one source genome to one or more target genomes.<br>
For this annotation set, CAT lifted over the reference annotations onto the T2T genome.<br>
It also lifted over the reference annotations onto these species using Liftoff and miniprot modules as well.<br>
CAT also incorporated Iso-Seq data, first assembled into transcripts with StringTie2, to make the final consensus annotation set.
</p>
</ul>
<h2>Credits</h2>
<p>
This track was provided by Prajna Hebbar <pnhebbar@ucsc.edu> of UC Santa Cruz.
</p>
<h2>References</h2>
<p>
	<li>Fiddes IT, Armstrong J, Diekhans M, Nachtweide S, Kronenberg ZN, Underwood JG, Gordon D, Earl D, Keane T, Eichler EE et al. Comparative Annotation Toolkit (CAT)-simultaneous clade and personal genome annotation. Genome Res. 2018 Jul;28(7):1029-1038. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/29884752/" target="_blank">29884752</a>
	<li>Stanke M, Diekhans M, Baertsch R, Haussler D. Using native and syntenically mapped cDNA alignments to improve de novo gene finding. Bioinformatics. 2008 Mar 1;24(5):637-44. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/18218656/" target="_blank">18218656</a>
	<li>Stanke M, Steinkamp R, Waack S, Morgenstern B. AUGUSTUS: a web server for gene finding in eukaryotes. Nucleic Acids Res. 2004 Jul 1;32(Web Server issue):W309-12. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/15215400/" target="_blank">15215400</a>
	<li>Armstrong J, Hickey G, Diekhans M, Fiddes IT, Novak AM, Deran A, Fang Q, Xie D, Feng S, Stiller J et al. Progressive Cactus is a multiple-genome aligner for the thousand-genome era. Nature. 2020 Nov;587(7833):246-251. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/33177663/" target="_blank">33177663</a>
	<li>Shumate A, Salzberg SL. Liftoff: accurate mapping of gene annotations. Bioinformatics. 2020 Dec 15;. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/33320174/" target="_blank">33320174</a>
	<li>Li, H. (2023) Protein-to-genome alignment with miniprot. Bioinformatics, 39, btad014; PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/36648328/" target="_blank">36648328</a>
</p>
