<h2>Description</h2>
<p>
Consensus gene models for the assembly generated using the CAT along with the NCBI RefSeq pipeline. Only protein coding genes are annotated in this consensus set.
</p>

<h2>Display Conventions and Configuration</h2>
<p>
This track follows the display conventions for gene prediction tracks.<br>
The exons for putative non-coding genes and untranslated regions are represented by relatively thin blocks, while those for coding open reading frames are thicker.
Gene names are displayed in 'pack' or 'full' mode. More information about each gene can be found by clicking on the specific gene/transcript model.<br><br>

The following color key is used:
</p>
<ul>
<li><span style='background-color:#0ABEFE;'>&nbsp;&nbsp;&nbsp;</span> - Protein Coding Genes
</ul>

<h2>Methods</h2>
<p>
This tracks displays gene annotations generated by finding a consensus between the annotations generated by the NCBI RefSeq pipeline and the Comparative Annotation Toolkit for each protein coding gene.
<ul>
<h3>Comparative Annotation Toolkit</h3>
Genome annotation for the assembly was performed using Comparative Annotation Toolkit (CAT). CAT leverages whole-genome alignments generated by Cactus to transfer annotations from one source genome to one or more target genomes. For this annotation set, CAT lifted over the reference annotations onto the T2T genome. It also lifted over the reference annotations onto these species using Liftoff and miniprot modules as well. CAT also incorporated Iso-Seq data, first assembled into transcripts with StringTie2, to make the final consensus annotation set.

<h3>NCBI RefSeq pipeline</h3>
The RefSeq annotation and RefSeq RNA alignment tracks were created at UCSC using data from the NCBI RefSeq project. GFF format data files were downloaded from the file GCF_029289425.2_NHGRI_mPanPan1-v2.0_pri_genomic.gff.gz delivered with the NCBI RefSeq genome assemblies at the <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/029/289/425/GCF_029289425.2_NHGRI_mPanPan1-v2.0_pri/" target="_blank">FTP location</a>. RefSeq annotations from GCF_029289425.2_NHGRI_mPanPan1-v2.0_pri were lifted to this GCA_029289425.2_NHGRI_mPanPan1-v2.0_pri assembly to provide these gene annotations on this corresponding assembly. The GFF file was converted to the genePred and PSL table formats for display in the Genome Browser. <br>
Same procedure has been performed for the other 5 primate genome primary assemblies, with entries GCA_029281585.2 for mGorGor1_v2.0_pri, GCA_028858775.2 for mPanTro3_v2.0_pri, GCA_028885655.2 for mPonAbe1_v2.0_pri, GCA_028885625.2 for mPonPyg1_v2.0_pri and GCA_028878055.2 for mSymSyn1_v2.0_pri.


<h3>Consensus Finding</h3>
To generate this, a reliable orthologous gene set was first generated. The orthologous genes were identified using the transMap method of CAT, which uses the Cactus alignment to map the orthologs. The cases where genes were mapped to a completely different neighborhood than the one in human were flagged and resolved using mappings from the liftoff mode. Then loci of all the protein-coding genes from this set were compared to the orthologous loci assigned by the NCBI RefSeq pipeline. For genes that were mapped to two completely different loci, the transcripts from both were mapped against human. Depending on the percentage identity of the generated protein to human (>50%), the transcripts were either discarded or assigned as ortholog/novel paralog. The novel gene loci that were annotated by either pipeline were collected and filtered on three levels: length of the protein generated >200 AA, identity with human protein >50%, and Iso-Seq transcript support. These were then merged into the consensus gene set.

</p>
</ul>
<h2>Credits</h2>
<p>
This track was provided by Prajna Hebbar <pnhebbar@ucsc.edu> of UC Santa Cruz. Data used to generate this track was also paritally produced at UCSC from data generated by scientists worldwide and curated by the NCBI RefSeq project.
</p>
<h2>References</h2>
<p>
<li>Fiddes IT, Armstrong J, Diekhans M, Nachtweide S, Kronenberg ZN, Underwood JG, Gordon D, Earl D, Keane T, Eichler EE et al. Comparative Annotation Toolkit (CAT)-simultaneous clade and personal genome annotation. Genome Res. 2018 Jul;28(7):1029-1038. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/29884752/" target="_blank">29884752</a>
<li>Stanke M, Diekhans M, Baertsch R, Haussler D. Using native and syntenically mapped cDNA alignments to improve de novo gene finding. Bioinformatics. 2008 Mar 1;24(5):637-44. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/18218656/" target="_blank">18218656</a>
<li>Stanke M, Steinkamp R, Waack S, Morgenstern B. AUGUSTUS: a web server for gene finding in eukaryotes. Nucleic Acids Res. 2004 Jul 1;32(Web Server issue):W309-12. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/15215400/" target="_blank">15215400</a>
<li>Armstrong J, Hickey G, Diekhans M, Fiddes IT, Novak AM, Deran A, Fang Q, Xie D, Feng S, Stiller J et al. Progressive Cactus is a multiple-genome aligner for the thousand-genome era. Nature. 2020 Nov;587(7833):246-251. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/33177663/" target="_blank">33177663</a>
<li>Shumate A, Salzberg SL. Liftoff: accurate mapping of gene annotations. Bioinformatics. 2020 Dec 15;. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/33320174/" target="_blank">33320174</a>
<li>Li, H. (2023) Protein-to-genome alignment with miniprot. Bioinformatics, 39, btad014; PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/36648328/" target="_blank">36648328</a>
<li>Kent WJ. BLAT - the BLAST-like alignment tool. Genome Res. 2002 Apr;12(4):656-64. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/11932250/" target="_blank">11932250</a>
<li>Pruitt KD, Brown GR, Hiatt SM, Thibaud-Nissen F, Astashyn A, Ermolaeva O, Farrell CM, Hart J, Landrum MJ, McGarvey KM et al. RefSeq: an update on mammalian reference sequences. Nucleic Acids Res. 2014 Jan;42(Database issue):D756-63. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/24259432/" target="_blank">24259432</a>
<li>Pruitt KD, Tatusova T, Maglott DR. NCBI Reference Sequence (RefSeq): a curated non-redundant sequence database of genomes, transcripts and proteins. Nucleic Acids Res. 2005 Jan 1;33(Database issue):D501-4. PMID: <a href="https://pmc.ncbi.nlm.nih.gov/articles/15608248/" target="_blank">15608248</a>
</p>
