<h2>Description</h2>
<p>Read coverage</p>

<h2>Methods</h2>
<p>
Input reads: HiFi and ONT reads (guppy 6.3.7~6.3.8)<br>
Mapper: Winnowmap v2.03 release (--MD -W repetitive_k15.txt -ax [map-pb | map-ont])<br>
Filter: Primary read alignments extracted with samtools view -F0x104<br>
</p>

<h2>Display Conventions and Configuration</h2>
<!-- if there are special conventions, like colors, describe them here -->
<p>
Pink: HiFi Primary alignments.<br>
Blue: ONT  Primary alignments.
</p>

<h2>Data access</h2>
<!-- Optional, links for downloading the data if there is a separate source beside the
  browser  -->
<p>Original bam files are available on <A HREF="https://genomeark.s3.amazonaws.com/index.html?prefix=species/Pan_troglodytes/mPanTro3/assembly_curated/mapping/">AWS GenomeArk</A>.
</p>

<h2>Release history</h2>
<ol>
  <li> 2024-01-04 Initial v2.0 release
</ol>

<h2>Contact</h2>
<p>
<li>Arang Rhie &lt;rhiea_at_nih.gov&gt;</a>
<li>Steven Solar &lt;steven.solar_at_nih.gov&gt;</a>
</p>

<h2>Credits</h2>
<ol>
  <li>Makova KD, Pickett BD, Harris RS, Hartley GA, Cechova M<em>et al</em>., The Complete Sequence and Comparative Analysis of Ape Sex Chromosomes, bioRxiv (2023). DOI: <a href="https://doi.org/10.1101/2023.11.30.569198" target="_blank" rel="noopener noreferrer">https://doi.org/10.1101/2023.11.30.569198</a>
</ol>
