<h2>Description</h2>
<p>
E/L Repli-seq data is a method to measure the replication timing of the genome. The data is generated by sequencing the DNA from early and late S-phase fractions. The ratio of the early to late fraction is used to determine the replication timing of the genome. The data is normalized to the genome-wide average and smoothed to remove noise. The data is represented as a log2 ratio of early to late fraction. Positive values indicate early replication, while negative values indicate late replication.
</p>

<h2>Methods</h2>
<p>
The Repli-seq data were processed in a similar way following the protocol (PMID: 29599440) and a similar pipeline available on this <a href="https://github.com/dvera/shart">Github repo</a>. The adapters of early and late fraction sequencing reads were removed using cutadapt (version 4.2) with parameters <code>-q 0 -O 1 -m 0</code> and adapter sequence AGATCGGAAGAGCACACGTCTG. The reads were then mapped to the primary assembly sequence using bwa mem (version 0.7.17) with the default options. PCR duplicates of the alignment were marked and removed using the samtool rmdup tool. We further cut the genome into non-overlapping 5kb bins and counted the number of filtered reads in each bin. The read count per bins was normalized into read per kilobase per million mapped reads (RPKM), and those bins with less then 0.1 RPKM across both early and late fraction are removed. Next, we calculated the log2 ratio between the early and late fraction.  The ratios were normalized using the interquartile range normalization, ie, (value - median)/IQR, where IQR represents the difference between the third quartile and the first quartile. Finally, the ratios of each chromosome are smoothed using the loess method with 300kb window size. <br>
The GM12878 was processed by re-mapping the data to T2T-CHM13v2. The original data was downloaded from 4DN data portal, with the experiement ID 4DNEXFBHF2ZI (https://data.4dnucleome.org/experiments-repliseq/4DNEXFBHF2ZI). HFF was processed similarly by download data from 4DN data portal with the experiment ID 4DNESJPG5AS9 (https://data.4dnucleome.org/experiment-set-replicates/4DNESJPG5AS9).
</p>

<h2>Credits</h2>
<p>
E/L Repli-seq data were generated by Takayo Sasaki in David Gilbert and Gabrielle Hartley and Emry Brannan in Rachel ONeill labs.
</p>

<p>
Repli-seq data processing was performed by Yang Zhang in Jian Ma lab in collaboration with David Gilbert lab.
</p>

<p>
The GM12878 and HFFc6 data has been generated by the 4DN lab. Please direct any questions to the Data Coordination and Integration Center. For more details, refer to the 4DN White Paper (doi:10.1038/nature23884) and the 4DN Data Portal paper (doi:10.1038/s41467-022-29697-4).
</p>

<p>
For inquiries, please contact us at the following address:
    <ul>
      <li> Yang Zhang: <a href="mailto:yangz6@andrew.cmu.edu">yangz6@andrew.cmu.edu</a></li>
    </ul>
</p>


